src/wallet/wallet.cpp

```diff
@@ -875,6 +875,30 @@ bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFlushOnClose)
     return true;
 }
 
+bool CWallet::RemoveFromWallet(const CTransaction& tx)
+{
+    LOCK(cs_wallet);
+
+    if (!WalletBatch(*database).EraseTx(tx.GetHash()))
+        return false;
+
+    WalletLogPrintf("RemoveFromWallet %s\n", tx.GetHash().ToString());
+
+    // Notify UI of new or updated transaction
+    NotifyTransactionChanged(this, tx.GetHash(), CT_DELETED);
+
+    // Notify an external script when a wallet transaction comes in or is updated
+    std::string strCmd = gArgs.GetArg("-walletnotify", "");
```

Comment:
I don't think this should use the same script.

Reply:
Yeah, that makes a lot of sense now that you point that out, however, I have no idea what I'd name that. Any ideas?

Reply:
Maybe remove notification? This is only happens after the RPC so the client doesn't need a notification.

Reply:
Yeah that's true